main.py

```
@@ -604,0 +1,1133 @@
+
+        # Check disk cache
+        cache_path = self._get_cache_path(url)
+        if cache_path.exists():
+            # Check if cache is expired
+            if time.time() - cache_path.stat().st_mtime <= self.max_age:
+                try:
+                    with open(cache_path, "r", encoding="utf-8") as f:
+                        content = f.read()
+                    # Add to memory cache
+                    self.memory_cache[url] = (content, time.time())
+                    return content
```

Ruff (0.8.2)

Consider moving this statement to an `else` block

(TRY300)